src/Magnum/DartIntegration/ConvertShapeNode.cpp

```diff
@@ -26,167 +26,319 @@
+            shapeData.meshes = Containers::Array<Trade::MeshData3D>(Containers::NoInit, 1);
+            new(&shapeData.meshes[0]) Trade::MeshData3D{Primitives::Capsule3D::solid(32, 32, 32, halfLength)};
+        }
+    } else if(shape->getType() == dart::dynamics::CylinderShape::getStaticType()) {
         auto cylinderShape = std::static_pointer_cast<dart::dynamics::CylinderShape>(shape);
 
         Float r = Float(cylinderShape->getRadius());
-        Float h = Float(cylinderShape->getHeight());
-        Float halfLength = 0.5f * h / r;
 
-        Trade::MeshData3D meshData{Primitives::Cylinder::solid(32, 32, halfLength)};
-        MeshTools::transformPointsInPlace(Matrix4::scaling(Vector3{r}), meshData.positions(0));
+        if(loadType & ConvertShapeType::Primitive)
+            shapeData.scaling = Vector3{r};
 
-        return ShapeData{std::move(meshData), std::move(matData), {}, {}};
-    }
-
-    if(shape->getType() == dart::dynamics::EllipsoidShape::getStaticType()) {
-        auto ellipsoidShape = std::static_pointer_cast<dart::dynamics::EllipsoidShape>(shape);
-
-        Eigen::Vector3d size = ellipsoidShape->getDiameters();
+        if(loadType & ConvertShapeType::Mesh) {
+            Float h = Float(cylinderShape->getHeight());
+            Float halfLength = 0.5f * h / r;
 
-        Trade::MeshData3D meshData{Primitives::Icosphere::solid(5)};
-        MeshTools::transformPointsInPlace(Matrix4::scaling(Vector3{Float(size(0)), Float(size(1)), Float(size(2))}), meshData.positions(0));
+            shapeData.meshes = Containers::Array<Trade::MeshData3D>(Containers::NoInit, 1);
+            new(&shapeData.meshes[0]) Trade::MeshData3D{Primitives::Cylinder::solid(32, 32, halfLength, Primitives::Cylinder::Flag::CapEnds)};
+        }
+    } else if(shape->getType() == dart::dynamics::EllipsoidShape::getStaticType()) {
+        if(loadType & ConvertShapeType::Primitive) {
+            auto ellipsoidShape = std::static_pointer_cast<dart::dynamics::EllipsoidShape>(shape);
 
-        return ShapeData{std::move(meshData), std::move(matData), {}, {}};
-    }
+            Eigen::Vector3d size = ellipsoidShape->getDiameters().array() * 0.5;
+            shapeData.scaling = Vector3(size(0), size(1), size(2));
+        }
 
-    if(shape->getType() == dart::dynamics::MeshShape::getStaticType()) {
-        /*  @todo check if we should not ignore the transformation in the Mesh */
+        if(loadType & ConvertShapeType::Mesh) {
+            shapeData.meshes = Containers::Array<Trade::MeshData3D>(Containers::NoInit, 1);
+            new(&shapeData.meshes[0]) Trade::MeshData3D{Primitives::Icosphere::solid(5)};
+        }
+    } else if(shape->getType() == dart::dynamics::MeshShape::getStaticType()) {
+        if (!importer) {
+            Error{} << Debug::boldColor(Debug::Color::Red) << "DartIntegration::convertShapeNode(): AssimpImporter is not available and you are trying to load a dart::dynamics::MeshShape" << Debug::resetColor;
```

Comment:
Unconditional output coloring will make problems when redirecting output to a text file (or on CIs that don't understand color codes). So I would rather have the boring uncolored output here instead.